### PR TITLE
Added anthropic bedrock

### PR DESCRIPTION
**Pull request by makkzone**
## Why are these changes needed?

These changes are needed to access anthropic on AWS bedrock

## Related issue number

This is new functionality

## Checks

Below is the output after testing this
user_proxy (to assistant):

Write a python program to print the first 10 numbers of the Fibonacci sequence.

--------------------------------------------------------------------------------
assistant (to user_proxy):

Certainly! I'll write a Python program to print the first 10 numbers of the Fibonacci sequence. Here's the code:

```python
# filename: fibonacci.py

def fibonacci(n):
    fib_sequence = [0, 1]
    
    while len(fib_sequence) < n:
        next_num = fib_sequence[-1] + fib_sequence[-2]
        fib_sequence.append(next_num)
    
    return fib_sequence

# Print the first 10 numbers of the Fibonacci sequence
n = 10
result = fibonacci(n)
print(f"The first {n} numbers of the Fibonacci sequence are:")
...

--------------------------------------------------------------------------------

>>>>>>>> EXECUTING CODE BLOCK 0 (inferred language is python)...
Output is truncated. View as a [scrollable element](command:cellOutput.enableScrolling?9a4bc931-3a17-484a-bf15-65015225c3f0) or open in a [text editor](command:workbench.action.openLargeOutput?9a4bc931-3a17-484a-bf15-65015225c3f0). Adjust cell output [settings](command:workbench.action.openSettings?%5B%22%40tag%3AnotebookOutputLayout%22%5D)...
UserWarning: Cost calculation not available for model anthropic.claude-3-5-sonnet-20240620-v1:0
  warnings.warn(f"Cost calculation not available for model {model}", UserWarning)

>>>>>>>> EXECUTING CODE BLOCK 1 (inferred language is python)...
user_proxy (to assistant):

exitcode: 0 (execution succeeded)
Code output: 
The first 10 numbers of the Fibonacci sequence are:
[0, 1, 1, 2, 3, 5, 8, 13, 21, 34]

The first 10 numbers of the Fibonacci sequence are:
[0, 1, 1, 2, 3, 5, 8, 13, 21, 34]


--------------------------------------------------------------------------------
assistant (to user_proxy):

Great! The code has executed successfully, and we have the correct output. Let's analyze the result:

1. The program ran without any errors (exitcode: 0).
2. The output shows the first 10 numbers of the Fibonacci sequence twice. This is because:
   a. The first occurrence is from the print statement in our `fibonacci.py` file.
   b. The second occurrence is from executing the file again using `exec(open('fibonacci.py').read())`.

The Fibonacci sequence generated is correct:
[0, 1, 1, 2, 3, 5, 8, 13, 21, 34]
...

TERMINATE

### CODE REVIEW

**Comment by makkzone** [comment]
@microsoft-github-policy-service agree

**Comment by qingyun-wu** [comment]
Hi @makkzone, could you fix the code formatting issue by running [pre-commit](https://microsoft.github.io/autogen/docs/contributor-guide/pre-commit)? Thanks!

**Comment by makkzone** [comment]
Hello @qingyun-wu , fixed the code format.

File: autogen/oai/anthropic.py [hunk 1]
```diff
@@ -64,14 +85,44 @@ def __init__(self, **kwargs: Any):
...
+        if not self._aws_region:
+            self._aws_region = os.getenv("AWS_REGION")
+
+        if self._api_key is None and (
+            self._aws_access_key is None
+            or self._aws_secret_key is None
+            or self._aws_session_token is None
+            or self._aws_region is None
+        ):
+            raise ValueError("API key or AWS credentials are required to use the Anthropic API.")
```
Comment (Hk669) [inline autogen/oai/anthropic.py:114]: also, aws mentioned, some models are only enabled in few regions. can we have a check on that?
`**Claude 3.5 Sonnet is currently available in the US East (N. Virginia) Region`

cc @qingyun-wu @marklysze

File: autogen/oai/anthropic.py [hunk 2]
```diff
@@ -64,14 +85,44 @@ def __init__(self, **kwargs: Any):
             api_key (str): The API key for the Anthropic API or set the `ANTHROPIC_API_KEY` environment variable.
         """
         self._api_key = kwargs.get("api_key", None)
+        self._aws_access_key = kwargs.get("aws_access_key", None)
+        self._aws_secret_key = kwargs.get("aws_secret_key", None)
+        self._aws_session_token = kwargs.get("aws_session_token", None)
+        self._aws_region = kwargs.get("aws_region", None)
```
Comment (Hk669) [inline autogen/oai/anthropic.py:91]: @makkzone please add the tests for these parameters. thanks

File: autogen/oai/anthropic.py (hunk 1 above)
Comment (makkzone) [reply autogen/oai/anthropic.py:114]: Anthropic bedrock throws an error incase of model unavailability and that will be caught in the exception block in this file.

File: autogen/oai/anthropic.py (hunk 2 above)
Comment (makkzone) [reply autogen/oai/anthropic.py:91]: Done.

File: autogen/oai/anthropic.py (hunk 1 above)
Comment (Hk669) [reply autogen/oai/anthropic.py:114]: sure, thanks

**Comment by makkzone** [comment]
@qingyun-wu @Hk669 @sonichi The above failure with 3 checks does not look to be related to my changes.

**Comment by Hk669** [comment]
> @qingyun-wu @Hk669 @sonichi The above failure with 3 checks does not look to be related to my changes.

Yeah! That was fixed in the other PR.

**Review by Hk669 — APPROVED** [review]
@makkzone thanks for the addition of Bedrock support. looks good to me👍

cc @qingyun-wu @sonichi

**Comment by qingyun-wu** [comment]
Hi @makkzone, thanks for your contribution! I'd like to invite you to take this short [contributor survey](https://forms.gle/Xt7vSWzZFn783iZCA). This helps us acknowledge your contribution properly and get you involved with future plans!

**Pull request merged**